EcosystemSampleApp/SampleAppViewController.swift

```diff
@@ -339,5 +342,61 @@ class SampleAppViewController: UIViewController, UITextFieldDelegate {
             }
         }
     }
+    
+    @IBAction func orderConfirmationTapped(_ sender: Any) {
+        guard let offerId = lastOfferId else{
+            
+            let alert = UIAlertController(title: "No Order has Been Made", message: "No Order was sent in this session yet.", preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "Oh ok", style: .cancel, handler: { [weak alert] action in
+                alert?.dismiss(animated: true, completion: nil)
+            }))
+            self.present(alert, animated: true, completion: nil)
+            
+            return
+        }
+        print("orderConfirmationTapped offerId : \(offerId)")
+        guard let appId = appId, let jwtPKey = privateKey else {
+            alertConfigIssue()
+            return
+        }
+        
+        do {
+            try jwtLoginWith(lastUser, id: appId)
+        } catch {
+            alertStartError(error)
+        }
+        
+        Kin.shared.orderConfirmation(for: offerId) { (status, err) in
+            DispatchQueue.main.async { [weak self] in
+                if let s = status {
+                    let statusMsg :String = String(reflecting :s)
+                    
+                    let alert = UIAlertController(title: nil, message: nil, preferredStyle: .alert)
+                    if let e = err {
+                        alert.title = "Failure"
+                        alert.message = "Native Earn failed: (\(e.localizedDescription))"
+                    } else {
+                        alert.title = "Get Order Confirmation - Success"
+                        alert.message = "status: \(statusMsg)\n"
+                        switch s{
+                        case .completed(let jwt):
+                            alert.addAction(UIAlertAction(title: "You can view the confirmation on jwt.io, View on jwt.io", style: .default, handler: { [weak alert] action in
```

Comment:
@yosriz does this fit? have you tested this?

Reply:
yes, that's the test I did here, in case the order already completed there's a jwt confirmation and you can observer it in jwt.io, o.w there's only status (pending/failed)